File: scripts/generate-icons.js
// Generates all Android mipmap launcher icons from logo/srklogo.png
// Also copies a 1024x1024 version to assets/images/icon.png for Metro.
// Run: npm run generate:icons
const path = require('path');
const sharp = require('sharp');

const ROOT = path.resolve(__dirname, '..');
const SRC  = path.join(ROOT, 'logo', 'srklogo.png');
const RES  = path.join(ROOT, 'android', 'app', 'src', 'main', 'res');

const SIZES = [
  { folder: 'mipmap-mdpi',    size: 48  },
  { folder: 'mipmap-hdpi',    size: 72  },
  { folder: 'mipmap-xhdpi',   size: 96  },
  { folder: 'mipmap-xxhdpi',  size: 144 },
  { folder: 'mipmap-xxxhdpi', size: 192 },
];

(async () => {
  // Metro require() source
  await sharp(SRC).resize(1024, 1024, { fit: 'cover' }).toFile(
    path.join(ROOT, 'assets', 'images', 'icon.png')
  );
  console.log('✓  assets/images/icon.png  1024x1024');

  for (const { folder, size } of SIZES) {
    const img = sharp(SRC).resize(size, size, { fit: 'cover' });
    await img.clone().toFile(path.join(RES, folder, 'ic_launcher.png'));
    await img.clone().toFile(path.join(RES, folder, 'ic_launcher_round.png'));
    console.log(`✓  ${folder}  ${size}x${size}`);
  }

  console.log('\nDone — rebuild with: npm run apk:release');
})();

File: scripts/generate_icons.py
"""
Generate all Android mipmap launcher icons from logo/srklogo.png.
Also copies a 1024x1024 version to assets/images/icon.png for Metro.

Requirements:
    pip install Pillow

Run from project root:
    python scripts/generate_icons.py
"""

import os
import shutil
from pathlib import Path
from PIL import Image

ROOT = Path(__file__).resolve().parent.parent
SRC  = ROOT / "logo" / "srklogo.png"
RES  = ROOT / "android" / "app" / "src" / "main" / "res"

SIZES = [
    ("mipmap-mdpi",    48),
    ("mipmap-hdpi",    72),
    ("mipmap-xhdpi",   96),
    ("mipmap-xxhdpi",  144),
    ("mipmap-xxxhdpi", 192),
]

def resize_and_save(src: Image.Image, dest: Path, size: int):
    img = src.copy()
    img = img.resize((size, size), Image.LANCZOS)
    dest.parent.mkdir(parents=True, exist_ok=True)
    img.save(dest, format="PNG", optimize=True)

def main():
    if not SRC.exists():
        print(f"ERROR: source logo not found at {SRC}")
        return

    with Image.open(SRC).convert("RGBA") as src_img:
        # Metro require() source
        metro_dest = ROOT / "assets" / "images" / "icon.png"
        metro_dest.parent.mkdir(parents=True, exist_ok=True)
        resize_and_save(src_img, metro_dest, 1024)
        print(f"✓  assets/images/icon.png  1024x1024")

        for folder, size in SIZES:
            resize_and_save(src_img, RES / folder / "ic_launcher.png",       size)
            resize_and_save(src_img, RES / folder / "ic_launcher_round.png", size)
            print(f"✓  {folder}  {size}x{size}")

    print("\nDone — rebuild with:  npm run apk:release")

if __name__ == "__main__":
    main()
